src/endpoint/s3/ops/s3_put_bucket_lifecycle.js

```diff
@@ -13,30 +13,42 @@ async function put_bucket_lifecycle(req) {
 
     // <Rule>
     //   <ID>id2</ID>
-    //   <Prefix>logs/</Prefix>
+    //   <Filter>
+    //     <Prefix>logs/</Prefix> 
+    //   </Filter>
     //   <Status>Enabled</Status>
     //   <Expiration>
     //     <Days>365</Days>
     //   </Expiration>
     // </Rule>
     const lifecycle_rules = _.map(req.body.LifecycleConfiguration.Rule, rule => {
-        var rule_id = uuid().split('-')[0];
+        let rule_id = uuid().split('-')[0];
         if (rule.ID) {
             rule_id = rule.ID[0];
         }
-        let current_rule = {
+        const current_rule = {
             id: rule_id,
-            prefix: rule.Prefix[0],
+            filter: {},
             status: rule.Status[0]
         };
+        if (rule.Prefix) {
+            throw new S3Error(S3Error.InvalidArgument);
+        }
+        if (rule.Filter) {
```

Comment:
what about all the other options in filter?  If we don't want to handle maybe we should throw something...

Reply:
Sure, We will throw `S3Error.NotImplemented`